app/src/main/java/org/mozilla/fenix/components/metrics/MetricsStorage.kt

```diff
@@ -64,27 +77,80 @@ internal class DefaultMetricsStorage(
             Event.GrowthData.FirstUriLoadForDay -> {
                 settings.uriLoadGrowthLastSent = System.currentTimeMillis()
             }
+            Event.GrowthData.FirstWeekSeriesActivity -> {
+                settings.firstWeekSeriesGrowthSent = true
+            }
         }
     }
 
+    private fun updateDaysOfUse() {
+        val daysOfUse = settings.firstWeekDaysOfUseGrowthData
+        val currentDate = Calendar.getInstance()
+        val asString = dateFormatter.format(currentDate.time)
+        if (currentDate.timeInMillis.withinFirstWeek() && daysOfUse.none { it == asString }) {
+            settings.firstWeekDaysOfUseGrowthData = daysOfUse.plus(asString)
+        }
+    }
+
+    private fun shouldTrackFirstWeekActivity(): Boolean = Result.runCatching {
+        if (!System.currentTimeMillis().withinFirstWeek() || settings.firstWeekSeriesGrowthSent) {
+            return false
+        }
+
+        val daysOfUse = settings.firstWeekDaysOfUseGrowthData.map {
+            dateFormatter.parse(it)
+        }.sorted()
```

Comment:
would we be able to save ourselves some work if we always save this data as sorted?

Reply:
Since SharedPreferences are saved as `Set`s I'm not sure we would have strong guarantees that the data would remain sorted. By sorting before use we can guarantee the dates are in the order we expect

Reply:
this is non-blocking, but I did find a handy function we could use if we wanted to relocate the sorting, `sortedSetOf(...)`

https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/sorted-set-of.html

I admittedly haven't spent a ton of time thinking if one location is optimal over the other, or if it matters.

Reply:
Oh this function definitely looks useful for the future! In the interest of avoiding bike-shedding an optimization of this size too much I think I will leave it for now. The size of the list will never be larger than a few elements.